Readiness probe failure on cmdline content

**Version**
OCI assets from the `latest` branch (98567b62effa1decd05a0a7b08d308b1111fb601) tested as part of AAP 2.4

**Describe the bug**
Since [1], the pulp type between api and content is determine by reading the `/proc/1/cmdline` file content within the container.

On API pod
```console
$ /usr/bin/readyz.py /api/galaxy/pulp/api/v3/status/
Traceback (most recent call last):
  File "/usr/bin/readyz.py", line 59, in <module>
    if pulp_type() == "[pulp-api]":
  File "/usr/bin/readyz.py", line 55, in pulp_type
    p_type = f.readline().split(" ")[2].strip("\x00")
IndexError: list index out of range
```
On Content pod
```console
$ /usr/bin/readyz.py /pulp/content/
Traceback (most recent call last):
  File "/usr/bin/readyz.py", line 59, in <module>
    if pulp_type() == "[pulp-api]":
  File "/usr/bin/readyz.py", line 55, in pulp_type
    p_type = f.readline().split(" ")[2].strip("\x00")
IndexError: list index out of range
```

However, the output of this file doesn't match what is described in the readyz.py script comment

> gunicorn: master \[pulp-{content,api,worker,resource-manager}\]

There's no space character in the cmdline file.
In fact, that pattern is only applicable for the kernel cmd (`/proc/cmdline`) but not the process cmdline (`/proc/xxx/cmdline`).

On API pod
```console
$ cat /proc/1/cmdline 
/usr/bin/python3.9/usr/bin/gunicorn--bind[::]:24817pulpcore.app.wsgi:application--namepulp-api--timeout90--workers2
```
On Content pod
```console
$ cat /proc/1/cmdline 
/usr/bin/python3.9/usr/bin/gunicornpulpcore.content:server--namepulp-content--bind[::]:24816--worker-classaiohttp.GunicornWebWorker--timeout90--workers2--access-logfile-
```

**To Reproduce**
Deploy pulp-operator with [1] included in the contain image.

**Expected behavior**
API and Content pods in ready state

**Additional context**

[1] https://github.com/pulp/pulp-oci-images/commit/98567b62effa1decd05a0a7b08d308b1111fb601

I was able to build and deploy AAP 2.4 with downstream images and deploy on an x86 cluster.  This is still from the ansible branch, I haven't switched to main for downstream builds yet.

I see this in the content pod logs:
```
[2023-01-26 22:06:54 +0000] [34] [ERROR] Exception in gunicorn worker
Traceback (most recent call last):
File "/usr/lib64/python3.9/site-packages/aiohttp/worker.py", line 57, in run
self.loop.run_until_complete(self._task)
File "/usr/lib64/python3.9/asyncio/base_events.py", line 647, in run_until_complete
return future.result()
File "/usr/lib64/python3.9/site-packages/aiohttp/worker.py", line 70, in _run
wsgi = await self.wsgi()
File "/usr/lib/python3.9/site-packages/pulpcore/content/__init__.py", line 65, in server
os.chdir(settings.WORKING_DIRECTORY)
FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/pulp/tmp'
```

And I see the error Dimitri reported with the readyz.py:

```
sh-4.4$ /usr/bin/readyz.py /pulp/content/
Traceback (most recent call last):
  File "/usr/bin/readyz.py", line 59, in <module>
    if pulp_type() == "[pulp-api]":
  File "/usr/bin/readyz.py", line 55, in pulp_type
    p_type = f.readline().split(" ")[2].strip("\x00")
IndexError: list index out of range
```

--- a/CHANGES/402.bugfix
+++ b/CHANGES/402.bugfix
@@ -1,0 +1,1 @@
+Fix readiness probe script based on gunicorn install method.

--- a/images/assets/readyz.py
+++ b/images/assets/readyz.py
@@ -44,20 +44,23 @@ def is_content_healthy(path):
 
 
 """
-Get container type based on entrypoint.
-Our entrypoint script is "exec'ing" (so no new process being created) the gunicorn process with the name of pulp "entity" (pulp-api, pulp-content, pulp-worker, pulp-resource-manager). Because of that, container's PID 1 will always be something like:
+PID 1 value (/proc/1/cmdline) may vary based on the gunicorn install method (RPM vs pip)
+The cmdline value for this PID looks like:
 ```
+# pip installation
 gunicorn: master \[pulp-{content,api,worker,resource-manager}\]
 ```
+OR
+```
+# RPM installation
+/usr/bin/python3.9/usr/bin/gunicorn--bind[::]:2481{6,7}pulpcore.app.wsgi:application--namepulp-{api,content}--timeout90--workers2
+```
 """
-def pulp_type():
-    f = open("/proc/1/cmdline", "r")
-    p_type = f.readline().split(" ")[2].strip("\x00")
-    f.close()
-    return p_type
+with open("/proc/1/cmdline", "r") as f:
+    cmdline = f.readline()
 
-if pulp_type() == "[pulp-api]":
+if "pulp-api" in cmdline:
     is_api_healthy(sys.argv[1])
 
-elif pulp_type() == "[pulp-content]":
+elif "pulp-content" in cmdline:
     is_content_healthy(sys.argv[1])

